### PR TITLE
Hackerspace login #346 side image

### LINKED ISSUE #345: SPIKE - Find new (copyright-free) image for login/signup pages
- To find an image that we have rights to use
- Even better: create an album of images that we can use

### LINKED ISSUE #346: UI Refresh - Login to Hackerspace
### Acceptance Criteria 
- Design to reflect a side image

### Links
Figma: https://www.figma.com/file/omAtzW5Kfg7NN97ECuQ3GC/Govhack-Hackerspace-Reskin?node-id=1080-948&t=hXhXKM8946xe3Atq-0

### PR DESCRIPTION
**Pull request by MunyaradziMagura**
### Pull Request Checklist

finished code for the login page of hackerspace, also edited the code for the register page to remove redundancy 

loginpagehackerspace

Fixes #346 
Fixes #345

### CODE REVIEW

File: app/assets/stylesheets/layouts/registration.scss [hunk 1]
```diff
@@ -144,6 +144,7 @@
             display: flex;
             margin: 50px 0px;
             width: 100%;
+
```
Comment (00Chaotic) [inline app/assets/stylesheets/layouts/registration.scss:147]: Nitpick: Remove extra line

File: app/assets/images/Overlay.png
Comment (00Chaotic) [inline app/assets/images/Overlay.png:1]: Is this confirmed to be a royalty-free image? Does the license require credit to be given to anyone or something similar?

**Comment by 00Chaotic** [comment]
The image of the login page in your comment makes it a little hard to see what the whole page looks like. Could you post another screenshot of the entire window so we can get a better idea?

File: app/assets/images/Overlay.png
Comment (bmpickford) [inline app/assets/images/Overlay.png:1]: Why the large increase in file size?

File: app/assets/stylesheets/layouts/registration.scss [hunk 2]
```diff
@@ -640,10 +641,13 @@
...
-          height: 90vh;
+          height: 100vh;
           width: 50%;
-          background-size: contain;
-          background: url("/assets/Overlay.png") no-repeat center;
+          background-size: cover;
+          background-image: url("/assets/Overlay.png");
+          background-position: center bottom;
+          background-repeat: no-repeat;
+
```
Comment (bmpickford) [inline app/assets/stylesheets/layouts/registration.scss:650]: nitpick:
```suggestion
```

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -31,5 +31,8 @@
         <% end %>
       </section>
     </section>
+    <section class="rhs-image">
+  </section>
```
Comment (bmpickford) [inline app/views/devise/sessions/new.html.erb:35]: nitpick: indent and extra line
```suggestion
    </section>
```

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -31,5 +31,8 @@
...
+
```
Comment (bmpickford) [inline app/views/devise/sessions/new.html.erb:36]:
```suggestion
```

File: app/assets/stylesheets/layouts/registration.scss [hunk 5]
```diff
@@ -640,10 +641,13 @@
 
         &.rhs-image {
           display: flex;
-          height: 90vh;
+          height: 100vh;
```
Comment (bmpickford) [inline app/assets/stylesheets/layouts/registration.scss:644]: why the change to 100vh? I kind of think this adds too much whitespace to the left hand side but if there's a case where it's needed happy to keep it

File: app/assets/images/Overlay.png
Comment (bmpickford) [reply app/assets/images/Overlay.png:1]: We do have another issue to replace this image: https://github.com/govhackaustralia/hackerspace3/issues/345

I'll see what I can find in WP and attach it to that issue. I don't mind if we combine them or keep them separate

**Review by MunyaradziMagura — COMMENTED** [review]
fixed changes, awaiting new image for addition

**Comment by MunyaradziMagura** [comment]
ive tested out all the images that Ben has provided in a previous commit and they all look fine. I was going to fetch the images directly from the govhack site however the images are not consistent (i.e. portrait pictures of people) which are not suitable for this use case. 

we can make a new issue about sorting out the images so they arnt hard coded. for now i feel like this is the best solution.


dynamic signin photo

File: app/assets/javascripts/images.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+
+// check if we are on the sign up / sign in page 
+if (window.location.href.endsWith("users/sign_in") || window.location.href.endsWith("/users/sign_up")){
```
Comment (00Chaotic) [inline app/assets/javascripts/images.js:3]: Will this check be run on every page? Is there a more efficient way to do this?

File: app/assets/images/Overlay.png
Comment (00Chaotic) [inline app/assets/images/Overlay.png:1]: Is this image still going to be used? Should this change be reverted, or is there a reason for the size change for this image?

File: app/assets/stylesheets/layouts/registration.scss [hunk 7]
```diff
@@ -642,8 +642,11 @@
           display: flex;
           height: 90vh;
           width: 50%;
-          background-size: contain;
-          background: url("/assets/Overlay.png") no-repeat center;
+          background-size: cover;
+          background-image: url("https://govhack.org/wp-content/uploads/2022/06/road-trip-with-raj-J61Sh5YrQho-unsplash.jpg");
```
Comment (00Chaotic) [inline app/assets/stylesheets/layouts/registration.scss:646]: We're setting background image through CSS here but then we also have the JS code to set it? Is there a need for both?

File: app/assets/images/Overlay.png
Comment (bmpickford) [inline app/assets/images/Overlay.png:1]: Why the large change in file size?

File: app/assets/javascripts/images.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    // list of images 
+    const imagesOverlays = [
```
Comment (bmpickford) [inline app/assets/javascripts/images.js:5]: this is a nice feature, but I think we want it to be a static image so we don't hit edge cases in sizing and cropping. I would also suggest downloading them and putting them under `app/assets/images/`, and maybe naming them something more descriptive of what they show.

Having them local also means that if they ever get deleted or moved in the wp site, we don't have to worry about them breaking.

File: app/assets/javascripts/images.js (hunk 6 above)
Comment (bmpickford) [inline app/assets/javascripts/images.js:3]: I'm not really all that familiar with how rails operates, but I think loading in javascript like this should be reserved for edge cases and should usually be done in ruby files.

File: app/assets/stylesheets/layouts/registration.scss (hunk 7 above)
Comment (MunyaradziMagura) [reply app/assets/stylesheets/layouts/registration.scss:646]: Yea we need both incase the end-user has js disabled on their browser it’ll have a default image to load

File: app/assets/javascripts/images.js
Comment (bmpickford) [inline app/assets/javascripts/images.js:1]: I can't see where this file is being loaded so not sure it's being used. If it's not being used can probably be deleted :)

File: app/assets/images/Overlay.png
Comment (MunyaradziMagura) [reply app/assets/images/Overlay.png:1]: Yes we can probably remove that

File: app/assets/javascripts/images.js (hunk 6 above)
Comment (MunyaradziMagura) [reply app/assets/javascripts/images.js:3]: Ideally I’d like it to run when that specific page opens but I’m not sure how to, if you have suggestions I’m all ears. I still haven’t gotten my head fully around how pages & files interact with one another

**Review by bmpickford — COMMENTED** [review]
Have the images as local assets would probably be better so it doesn't break if/when we decide to change the location or name of the images in wordpress.

If the old image isn't being used, it can probably also be deleted. If you get stuck or need a hand feel free to reach out!

File: app/assets/stylesheets/layouts/registration.scss [hunk 9]
```diff
@@ -144,7 +144,7 @@
             display: flex;
             margin: 50px 0px;
             width: 100%;
-
+
```
Comment (bmpickford) [inline app/assets/stylesheets/layouts/registration.scss:147]: nitpick: remove extra whitespace
```suggestion
```

**Review by bmpickford — APPROVED** [review]
Looks good to me!
